saw/Main.hs

```diff
@@ -36,15 +38,20 @@ main = do
       case files of
         _ | showVersion opts'' -> hPutStrLn stderr shortVersionText
         _ | showHelp opts'' -> err opts'' (usageInfo header options)
-        [] -> REPL.run opts''
-        _ | runInteractively opts'' -> REPL.run opts''
-        [file] -> processFile (AIGProxy GIA.proxy) opts'' file `catch`
-                  (\(ErrorCall msg) -> err opts'' msg)
+        [] -> checkZ3 opts'' *> REPL.run opts''
+        _ | runInteractively opts'' -> checkZ3 opts'' *> REPL.run opts''
+        [file] -> checkZ3 opts'' *>
+          processFile (AIGProxy GIA.proxy) opts'' file `catch`
+          (\(ErrorCall msg) -> err opts'' msg)
         (_:_) -> err opts'' "Multiple files not yet supported."
     (_, _, errs) -> do hPutStrLn stderr (concat errs ++ usageInfo header options)
                        exitProofUnknown
   where header = "Usage: saw [OPTION...] [-I | file]"
+        checkZ3 opts = do
+          p <- findExecutable "z3"
```

Comment:
is `z3` the only solver here?

Reply:
Z3 is the only one that's absolutely required for the Cryptol typechecker, I think